openwisp_notifications/base/models.py

```diff
@@ -105,36 +105,74 @@ def target_url(self):
     def message(self):
         return self.get_message()
 
+    @cached_property
+    def rendered_description(self):
+        if not self.description:
+            return
+        md_description = self._get_formatted_string(self.description)
+        return mark_safe(markdown(md_description))
+
     @property
     def email_message(self):
         return self.get_message(email_message=True)
 
+    def _set_extra_attributes(self, email_message=False):
```

Comment:
I thought of performing these operations in the `__init__`, but the target_link is dynamic based on whether we are using the message in the email.